src/coreclr/src/utilcode/sstring.cpp

```diff
@@ -197,49 +169,21 @@ int SString::CaseCompareHelperA(const CHAR *buffer1, const CHAR *buffer2, COUNT_
     {
         CHAR ch1 = *buffer1;
         CHAR ch2 = *buffer2;
-        if ((ch1 == 0) || (ch2 == 0))
+        diff = ch1 - ch2;
+        if  (diff != 0 || stopOnNull)
         {
-            diff = ch1 - ch2;
-            if  (diff != 0 || stopOnNull)
+            if (!CAN_SIMPLE_UPCASE_ANSI(ch1) || !CAN_SIMPLE_UPCASE_ANSI(ch2))
```

Comment:
I would delete this condition - it will early for characters <0x20. I think it is better to always take the `SIMPLE_UPCASE_ANSI` path here.